File: daw24_HW1/Python/hw1.py
#################################################################
# Program reads in an input file(input.txt - a plain text file	#
# which contains 5 sentances), line by line anc check if the	#
# line read in is a pangram or not. If the sentence read is a	#
# pangram, it writes 'true' to the output file. If it's not, it #
# writes 'false' to the output file.				#
# Author: D. Weidenborner					#
# 1/12/2016							#
#################################################################

import string

# Checks letters from the alphabet against each sentance
def check(sent):
	letters = []
	for letter in sent:
		letters.append(letter)
	for letter in string.ascii_lowercase:
		if letter not in sent:
			return 0
	return 1

def hw1(input, output):
	# Read input file line by line
	f = open(input, 'r')
	strings = []
	for line in f:
		strings.append(line)
	f.close()
	
	# For every line in input file, check if it contains
	# all alphabets - store result (true/false) in a variable
	answers = []
	for line in strings:
		if (len(line) < 27):
			answers.append('false')
		elif (check(line) == 1):
			# Call helper function to check against alphabet
			answers.append('true')
		else:
			answers.append('false')
	
	# Write result to output file
	f = open(output, 'w')
	for i in answers:
		f.write(i)
		f.write('\n')
	f.close()

# For testing
#hw1('input.txt','output.txt')

File: daw24_HW1/Java/hw1.java
/*
#################################################################
# Program reads in an input file(input.txt - a plain text file  #
# which contains 5 sentances), line by line anc check if the    #
# line read in is a pangram or not. If the sentence read is a   #
# pangram, it writes 'true' to the output file. If it's not, it #
# writes 'false' to the output file.                            #
# Author: D. Weidenborner                                       #
# 1/15/2016                                                     #
#################################################################
*/

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Scanner;
import java.util.ArrayList;
import java.io.FileWriter;

public class hw1{
	public static void hw1(String inFile, String outFile)throws FileNotFoundException, IOException{
		//Function defined here
		ArrayList<String> lines = new ArrayList<String>();
		File inputFile = new File(inFile);
		Scanner in = new Scanner(inputFile);
		while (in.hasNextLine()){ 
			String line = in.nextLine();
			lines.add(line);
		}
		in.close()
;
		ArrayList<String> letters = new ArrayList<String>();
		ArrayList<String> answers = new ArrayList<String>();
		// For each string
		for(int i = 0; i < lines.size(); i++){
			// Flag to check if a letter was not found
			// Set to 1 when letter not found so value is not set to true
			int flag = 0;
			// For each letter a-z lowercase
			for(char letter = 'a'; letter <= 'z'; letter++){
				int num = lines.get(i).indexOf(letter);
				// Returns -1 if letter not found
				if(num < 0){
					answers.add("false");
					flag = 1;
					break;
				}
			}
			// If flag is still zero after itterating through string it is a pangram
			if(flag == 0){
				answers.add("true");
			}
			// If flag was raised then answer was already set to false
			// Reset flag and continue to next string
			else if(flag == 1){
				flag = 0;
			}	
		}
		FileWriter out = new FileWriter(outFile);				
		for(String answer : answers){
			out.write(answer + "\n");
		}
		out.close();
	}
}
